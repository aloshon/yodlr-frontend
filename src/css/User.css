.user-card-body.card-body {
    color: "#AAAAAA";
}

.user-container.container {
   position: relative;
   margin: 10px;
 
}

.text-center.user-card.card {
    background-color: #AAAAAA;
}


@media (min-width: 800px) {
    .user-container.container {
        display: inline-block;
        width: 45%;
        vertical-align: top;
    }
}

.card-link:hover {
    cursor: pointer;
}