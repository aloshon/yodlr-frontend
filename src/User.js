import React, {useState, useContext} from "react";
import Card from "react-bootstrap/Card";
import Container from "react-bootstrap/Container";
import "./css/User.css";
import StateContext from "./StateContext";

/**
 * User renders all user info and function to toggle state
 * between "active" and "pending"
 */
const User = ({id, firstName, lastName, email, state, isAdmin}) => {
    const [userState, setUserState] = useState(state);
    const {updateState} = useContext(StateContext);

    const toggleState = async () => {
        // toggle state and then post it on api
        let toggledState = userState === "active" ? "pending" : "active";
        let newState = await updateState({id, firstName, lastName, email, state: toggledState, isAdmin});

        // update new state
        setUserState(newState.state);
    }

    return (
        <Container className="user-container">
            <Card className="text-center user-card" >
                <Card.Body className="user-card-body">
                    <Card.Title>{firstName} {lastName}</Card.Title>
                    <Card.Text><small>{email}</small></Card.Text>
                    <Card.Link
                    style={{color: userState === "active" ? "green" : "red"}}
                    onClick={toggleState}>
                        <small>{userState}</small>
                    </Card.Link>
                </Card.Body>
            </Card>
        </Container>
    )
}

export default User;