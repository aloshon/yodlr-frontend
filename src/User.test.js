import { render } from '@testing-library/react';
import User from './User';
import {Router} from 'react-router-dom';
import { createMemoryHistory } from 'history';
import StateContext from "./StateContext";

describe("User should be rendered", () => {
    const history = createMemoryHistory();
    const {asFragment, getByText} = render(
        <Router history={history}>
            <StateContext.Provider value={{updateState: jest.fn()}}>
                <User 
                id={1}
                firstName={"Testing"}
                lastName={"This"}
                email={"thisis@test.com"}
                state={"pending"}
                update={jest.fn}
                />
            </StateContext.Provider>
        </Router>
    );


    test('Load user data', () => {
        const fullName = getByText("Testing This");
        const email = getByText("thisis@test.com");
        const state = getByText("pending");
        expect(fullName).toBeInTheDocument();
        expect(email).toBeInTheDocument();
        expect(state).toBeInTheDocument();;
    });
    
    test("matches snapshot", () => {
        expect(asFragment()).toMatchSnapshot();
    });
});