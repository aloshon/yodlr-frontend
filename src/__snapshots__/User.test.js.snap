// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should be rendered matches snapshot 1`] = `<DocumentFragment />`;
